scripts/perf/perf_kit/retry_and_time.py

```diff
@@ -0,0 +1,133 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import contextlib
+import functools
+import math
+import random
+import signal
+import time
+
+
+class TimingResult:
+    def __init__(self):
+        self.start_time = 0
+        self.end_time = 0
+        self.value = 0
+
+
+@contextlib.contextmanager
+def timing(retry_count=1):
+    start_time = time.time()
+    result = TimingResult
+    try:
+        yield result
+    finally:
+        end_time = time.time()
+        diff = (end_time - start_time) * 1000.0
+
+        result.start = start_time
+        result.end_time = end_time
+        if retry_count == 1:
+            result.value = diff
+            print(f"Retry time: {diff:.3f} ms")
+        else:
+            average_time = diff / retry_count
+            result.value = average_time
+            print(f"Average time: {average_time:.3f} ms")
+
+
+def retry(retry_count=5):
```

Comment:
```suggestion
def repeat(retry_count=5):
```

Retry usually means "try up to N times, but stop when we get a success" -- this always runs the given number of times.